### PR TITLE
GUI-2839: Allow broken cookie to be cleared (following console restart) when GovCloud region is mistakenly enabled

### PR DESCRIPTION
**Pull request by kamalgill**
Also, prevent duplicate "US GovCloud" entries from displaying in region choices.

https://eucalyptus.atlassian.net/browse/GUI-2839

### CODE REVIEW

File: eucaconsole/views/dashboard.py [hunk 1]
```diff
@@ -69,6 +71,11 @@ def __init__(self, request):
 
     @view_config(route_name='dashboard', request_method='GET', renderer='../templates/dashboard.pt')
     def dashboard_home(self):
+        if self.conn is None:
+            # Handle broken connection case (e.g. configured AWS GovCloud region without proper access)
+            forget(self.request)
```
Comment (dkavanagh) [inline eucaconsole/views/dashboard.py]: I'm not sure you're using forget() as intended: http://docs.pylonsproject.org/projects/pyramid/en/latest/api/security.html#pyramid.security.forget

File: eucaconsole/layout.py [hunk 2]
```diff
@@ -73,11 +73,13 @@ def __init__(self, context, request):
         self.default_region = ''
         if self.cloud_type == 'aws':
             self.regions = AWS_REGIONS
+            govcloud_region_option = dict(
+                name='us-gov-west-1',
+                label='US GovCloud',
+            )
             if asbool(request.registry.settings.get('aws.govcloud.enabled', 'false')):
```
Comment (dkavanagh) [inline eucaconsole/layout.py]: Curious why this change? It creates the dict in all cases as compared to the old code that only created it when needed.

File: eucaconsole/views/dashboard.py (hunk 1 above)
Comment (dkavanagh) [inline eucaconsole/views/dashboard.py]: re-reading this and I noticed the additional condition. Was the gov cloud being appended multiple times?

Comment (kamalgill) [inline eucaconsole/views/dashboard.py]: Yes, the GovCloud label was appended multiple times.  Regarding the forget() method, we need that to clear the cookie to avoid a persistent internal server error.

File: eucaconsole/layout.py (hunk 2 above)
Comment (kamalgill) [inline eucaconsole/layout.py]: Good catch, I'll move the dict inside the aws.govcloud.enabled check.

File: eucaconsole/views/dashboard.py (hunk 1 above)
Comment (dkavanagh) [inline eucaconsole/views/dashboard.py]: OK, It wasn't clear to me that the forget method had that side effect.
